back-end/src/sciler/handler/handler.go

```diff
@@ -80,25 +80,99 @@ func (handler *Handler) onConnectionMsg(raw Message) {
 		logrus.Info("connection message received from: ", raw.DeviceID)
 		value, ok2 := contents["connection"]
 		if !ok2 || reflect.TypeOf(value).Kind() != reflect.Bool {
-			logrus.Error("received improperly structured connection message from device " + raw.DeviceID)
+			logrus.Warn("received improperly structured connection message from device " + raw.DeviceID)
+			return
 		}
 		device.Connection = value.(bool)
 		handler.Config.Devices[raw.DeviceID] = device
 		handler.SendStatus(raw.DeviceID)
 	}
 }
 
+// checkStatusType checks if the type of the status change is correct for the component
+func (handler *Handler) checkStatusType(device config.Device, v interface{}, k string) error {
+	valueType := reflect.TypeOf(v).Kind()
+	if inputType, ok := device.Input[k]; ok {
```

Comment:
can't this be simplified to take the device input/output as method input and calling it twice instead of double if?